src/ToolBox/SOS/Strike/util.cpp

```diff
@@ -6185,184 +5975,93 @@ HRESULT __stdcall PERvaMemoryReader::ReadExecutableAtRVA(DWORD relativeVirtualAd
+    // TODO: in-memory windows PDB not supported
+    hr = LoadSymbolsForWindows(pMD, moduleData.LoadedPEAddress, pModuleName, moduleData.bIsInMemory);
+    if (SUCCEEDED(hr))
     {
-        ExtErr("Error: Fail to initialize CoreCLR\n");
-        return Status;
+        return hr;
     }
+#endif // FEATURE_PAL
 
-    coreclr_create_delegate_ptr CreateDelegate =
-        (coreclr_create_delegate_ptr)dlsym(coreclrLib, "coreclr_create_delegate");
-
-    IfFailRet(CreateDelegate(hostHandle, domainId, SymbolReaderDllName,
-                        SymbolReaderClassName, "ResolveSequencePoint",
-                        (void **)&resolveSequencePointDelegate));
-    IfFailRet(CreateDelegate(hostHandle, domainId, SymbolReaderDllName,
-                        SymbolReaderClassName, "LoadSymbolsForModule",
-                        (void **)&loadSymbolsForModuleDelegate));
-    IfFailRet(CreateDelegate(hostHandle, domainId, SymbolReaderDllName,
-                        SymbolReaderClassName, "GetLocalVariableName",
-                             (void **)&getLocalVariableNameDelegate));
-    IfFailRet(CreateDelegate(hostHandle, domainId, SymbolReaderDllName,
-                        SymbolReaderClassName, "GetLineByILOffset",
-                             (void **)&getLineByILOffsetDelegate));
-    return Status;
+    return LoadSymbolsForPortablePDB(
+        moduleData.bIsInMemory,
+        pModuleName, 
+        moduleData.LoadedPEAddress,
+        moduleData.LoadedPESize, 
+        moduleData.InMemoryPdbAddress,
+        moduleData.InMemoryPdbSize);
 }
 
-HRESULT SymbolReader::GetLineByILOffset(__in_z const char* szModuleName, mdMethodDef MethodToken,
-                                         ULONG64 IlOffset, ___out ULONG *pLinenum,
-                                         __out_ecount(cbFileName) LPSTR lpszFileName,
-                                         ___in ULONG cbFileName)
-{
-    HRESULT Status = S_OK;
-
-    if (getLineByILOffsetDelegate == nullptr)
-    {
-        IfFailRet(SymbolReader::PrepareSymbolReader());
-    }
-
-    BSTR wszFileName = SysAllocStringLen(0, cbFileName);
-    if (wszFileName == nullptr)
-    {
-        return E_OUTOFMEMORY;
-    }
-
-    if (getLineByILOffsetDelegate(szModuleName, MethodToken, IlOffset, pLinenum, &wszFileName) == TRUE)
-    {
-        WideCharToMultiByte(CP_ACP, 0, wszFileName, (int) (_wcslen(wszFileName) + 1),
-                            lpszFileName, cbFileName, NULL, NULL);
-        if (*pLinenum == 0)
-        {
-            *pLinenum = -1;
-            Status = E_FAIL;
-        }
-    }
-    else
-    {
-        Status = E_FAIL;
-    }
-
-    SysFreeString(wszFileName);
-
-    return Status;
-}
-#endif //FEATURE_PAL
+#ifndef FEATURE_PAL
 
-HRESULT SymbolReader::LoadSymbols(IMetaDataImport * pMD, ULONG64 baseAddress, __in_z WCHAR* pModuleName, BOOL isInMemory)
+HRESULT SymbolReader::LoadSymbolsForWindows(IMetaDataImport* pMD, ULONG64 peAddress, __in_z WCHAR* pModuleName, BOOL isInMemory)
```

Comment:
LoadSymbolsForRegularPDBs?

Sent from my Verizon, Samsung Galaxy smartphone


-------- Original message --------
From: Aditya Mandaleeka <notifications@github.com>
Date: 8/16/16 12:47 PM (GMT-08:00)
To: dotnet/coreclr <coreclr@noreply.github.com>
Cc: Mike McLaughlin <mikem@microsoft.com>, Assign <assign@noreply.github.com>
Subject: Re: [dotnet/coreclr] Change SOS to use wrapper class that is now in System.Diagnostics.StackTrace. (#6751)


In src/ToolBox/SOS/Strike/util.cpp<https://github.com/dotnet/coreclr/pull/6751#discussion_r75006303>:

>
> -HRESULT SymbolReader::LoadSymbols(IMetaDataImport * pMD, ULONG64 baseAddress, __in_z WCHAR* pModuleName, BOOL isInMemory)
> +HRESULT SymbolReader::LoadSymbolsForWindows(IMetaDataImport* pMD, ULONG64 peAddress, __in_z WCHAR* pModuleName, BOOL isInMemory)


I'm not a huge fan of the name LoadSymbolsForWindows since it sounds like it literally loads symbols for Windows code.

-
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://github.com/dotnet/coreclr/pull/6751/files/390282dcd0f548cd0dfda73f9e660758d9c8d9a7#r75006303>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AJ-Fa-FX3yH4_ksxicK8XpYNDNP5VxQ4ks5qghPKgaJpZM4JlCna>.